### PR TITLE
stop using deprecated api in feature tests

### PR DESCRIPTION
**Pull request by charlierudolph**
Currently our CI output includes this warning:

```
(node:2090) DeprecationWarning: `Cli` is deprecated, use `runCucumber` instead; see https://github.com/cucumber/cucumber-js/blob/main/docs/deprecations.md
(Use `node --trace-deprecation ...` to show where the warning was created)
```

Updated cucumber-js feature tests to stop using our own deprecated API

### CODE REVIEW

File: features/support/world.ts [hunk 1]
```diff
@@ -83,16 +85,20 @@ export class World {
...
+          provided: argvConfiguration,
+        },
+        environment
+      )
       let error: any
       try {
-        const { success } = await cli.run()
+        const { success } = await runCucumber(runConfiguration, environment)
         if (!success) {
           error = new Error('CLI exited with non-zero')
```
Comment (davidjgoss) [inline features/support/world.ts:103]: Should this now say something like "runCucumber threw an error"?

File: features/support/world.ts [hunk 2]
```diff
@@ -83,16 +85,20 @@ export class World {
...
       const stderr = new PassThrough()
-      const cli = new Cli({
-        argv: args,
-        cwd,
-        stdout,
-        stderr,
-        env,
-      })
+      const environment: IRunEnvironment = { cwd, stdout, stderr, env }
+      const { options, configuration: argvConfiguration } =
```
Comment (davidjgoss) [inline features/support/world.ts:89]: Looking at the 2 failing scenarios, I think this call to `parse` and the one below to `loadConfiguration` should also move inside of the try block.

**Review by davidjgoss — COMMENTED** [review]
Looks good, couple of suggestions.

**Comment by davidjgoss** [comment]
When first implementing `runCucumber` I looked at this and half-considered just making `@spawn` the only behaviour i.e. so they all run via the executable, but that would probably slow it down a lot.

File: features/support/world.ts (hunk 1 above)
Comment (charlierudolph) [reply features/support/world.ts]: Updated

File: features/support/world.ts (hunk 2 above)
Comment (charlierudolph) [reply features/support/world.ts]: Moved inside try block

**Review by davidjgoss — APPROVED** [review]

**Pull request merged**